### PR TITLE
[GH-584] Improved function specs

### LINKED ISSUE #584: Improve function specs
Go over specs and check if they can be more specific, fill in missing specs for public functions

### PR DESCRIPTION
**Pull request by DanielaIvanova**
resolves #584

### CODE REVIEW

**Comment by DanielaIvanova** [comment]
ok, I will check

**Comment by thepiwo** [comment]
@d-velev @gspasov what is your opinion on the `start_link` specs? I don't think they are needed to have.

File: apps/aecore/lib/aecore/tx/pool/worker.ex [hunk 1]
```diff
@@ -18,6 +18,7 @@ defmodule Aecore.Tx.Pool.Worker do
 
   @type tx_pool :: map()
 
+  @spec start_link(any()) :: :ignore | {:error, any()} | {:ok, pid()}
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/tx/pool/worker.ex]: this looks better than the huge spec used in the other `start_link`

Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/tx/pool/worker.ex]: I can remove all `start_link` specs if you think that is better without them. I really don't like it!

Comment (thepiwo) [reply apps/aecore/lib/aecore/tx/pool/worker.ex]: thank you

**Pull request merged**